stance/skeleton/skeleton.py

```diff
@@ -1,43 +1,71 @@
 import numpy as np
-from collections import namedtuple
-
-COCOPts = namedtuple('COCOPts', ('nose',
-                                 'neck',
-                                 'right_shoulder',
-                                 'right_elbow',
-                                 'right_wrist',
-                                 'left_shoulder',
-                                 'left_elbow',
-                                 'left_wrist',
-                                 'right_hip',
-                                 'right_knee',
-                                 'right_ankle',
-                                 'left_hip',
-                                 'left_knee',
-                                 'left_ankle'))
-
-
-class Skeleton:
-    def __init__(self, coco_points):
+from typing import Dict, Tuple, Sequence
+
+
+COCOPts = Sequence[Tuple[float, float]]
+
+
+class SKVector(object):
+    def __init__(self, head: Tuple[float, float], tail: Tuple[float, float]) -> None:
+        self.head: np.ndarray = np.array(head)
+        self.tail: np.ndarray = np.array(tail)
+        self.vec = self.head - self.tail
+
+
+    def cos_similarity(self, other: 'SKVector') -> float:
+        """Computes the cosine similarity between this vector and another
+
+        Parameters
+        ----------
+        other : SKVector
+            another vector
+
+        Returns
+        -------
+        float
+            cosine similarity between self and other with higher values being
+            more similar
+        """
+        magnitude = np.linalg.norm(self.vec) * np.linalg.norm(other.vec)
+        return np.dot(self.vec, other.vec) / magnitude
+
+
+    def __repr__(self) -> str:
+        return 'SKVector: {} -> {}'.format(tuple(self.head), tuple(self.tail))
+
+
+class Skeleton(object):
+    def __init__(self, coco_points: COCOPts) -> None:
         """
         Uses the output from the COCO model to create body variables
 
         Parameters
         ----------
-        coco_points : List[Points]
+        coco_points : COCOPts
             points in the COCO format
         """
-        self.body_points = COCOPts(coco_points)
+        self.body_points = coco_points
+        self.vectors: Dict[str, SKVector] = {
+            'l_lower_leg': SKVector(head=coco_points[13], tail=coco_points[12]),
```

Comment:
Do you think it's worth putting in this PR?

Reply:
I wouldn't want to forget about it, and it should just be a couple more lines. Plus it would be better to complete the whole class at once and not have to come back to it. So ya, pls include.